### PR TITLE
Sad path mountain project api

### LINKED ISSUE #53: Issue: Fix Climbing Drop Downs
5.10 displaying as 5.1

### PR DESCRIPTION
**Pull request by madelynrr**
**Functionality:**
-Adds sad path testing for climb params page
-Change max distance to numeric test field
-Add if/else block to escape facade to build climbs if no type specified
-Add if/else logic in view to display different message if no climbs returned
-Deleted old 404 error in escape service that wasn't being tested or hit
-Add logic to check that max difficulty is greater than min difficulty

**Testing:**
-Testing at 100% and passed Travis

**Related Issues:**
Closes #53

### CODE REVIEW

File: app/poros/escape_facade.rb [hunk 1]
```diff
@@ -2,14 +2,18 @@ class EscapeFacade
...
 
   def climb_options
     results = EscapeService.new.get_climb_results(@params, @location)
     results.map do |result|
-      if result[:type] == @params[:type] || result[:type] == "Sport, TR"
+      if result[:type] == @params[:trad] || result[:type] == "Sport, TR"
+        ClimbOptions.new(result)
+      elsif result[:type] == @params[:sport] || result[:type] == "Sport, TR"
+        ClimbOptions.new(result)
+      elsif !@params.key?(:trad) && !@params.key?(:sport)
```
Comment (tyladevon) [inline app/poros/escape_facade.rb]: Would like to discuss this change and get clarification on why we are looking to instantiate ClimbOptions twice here.

File: app/views/search/climbs/new.html.erb [hunk 2]
```diff
@@ -7,10 +7,13 @@
...
   <br>
-  <%= select_tag(:type, options_for_select(climb_type)) %>
+  <%= check_box_tag(:trad, "Trad") %>
+  <%= label_tag(:type, "Trad") %>
+  <%= check_box_tag(:sport, "Sport") %>
+  <%= label_tag(:type, "Sport") %>
   <br>
   <%= label_tag :distance %>
-  <%= text_field_tag :distance %>
+  <%= number_field_tag :distance %>
```
Comment (tyladevon) [inline app/views/search/climbs/new.html.erb]: Nice change here for number_field_tag

File: app/views/trip/new.html.erb [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
+      <p><%= climb.rating %></p>
+      <p><%= climb.stars %></p>
+      <p><%= climb.pitches %></p>
+      <p><%= climb.location %></p>
+      <p><%= climb.url %></p>
+    </section>
+  <% end %>
+<% else %>
+  <h4>No climbs exist for those parameters.</h4>
+  <%= button_to "Try Again", "/search/climbs/new", method: :get %>
```
Comment (tyladevon) [inline app/views/trip/new.html.erb]: Great sad path implementation here for scenarios resulting in receiving no results.

File: spec/features/users/can_view_search_results_spec.rb [hunk 4]
```diff
@@ -37,13 +37,13 @@
 
       select "5.7", from: :min_diff
       select "5.12", from: :max_diff
-      select "Sport", from: :type
+      check('sport')
```
Comment (tyladevon) [inline spec/features/users/can_view_search_results_spec.rb]: Nice change to check boxes.

**Comment by tyladevon** [comment]
I approve this PR.

**Comment by rallen20** [comment]
Closing pull request, all sad paths are included in a different Pull Request.

**Pull request closed without merging**